net/blockwise/blockwise.go

```diff
@@ -482,10 +484,11 @@ func (b *BlockWise) RemoveFromResponseCache(token message.Token) {
 	b.sendingMessagesCache.Delete(token.Hash())
 }
 
-func (b *BlockWise) sendEntityIncomplete(w ResponseWriter, token message.Token) {
+func (b *BlockWise) sendEntityIncomplete(w ResponseWriter, r Message, token message.Token) {
 	sendMessage := b.acquireMessage(w.Message().Context())
 	sendMessage.SetCode(codes.RequestEntityIncomplete)
 	sendMessage.SetToken(token)
+	setTypeFrom(sendMessage, r)
```

Comment:
This is not OK. You cannot send  response with `Confirmable` for `Confirmable` request.  I don't think you need to  add this line.

Reply:
Ok, maybe it is safer to always set it to `NonConfirmable` then? According to the spec `Confirmable` responses are allowed, but maybe they are unsuitable in this implementation? `NonConfirmable` responses are also always allowed if I read the spec correctly.

https://www.rfc-editor.org/rfc/rfc7252#section-5.2.3

Reply:
Yeh, this is fine. I'm preparing the new go-coap/v3 in https://github.com/plgd-dev/go-coap/issues/368 for the October release. Please is it possible to migrate this fix to branch [v3](https://github.com/plgd-dev/go-coap/tree/v3)?

Reply:
@jkralik Changed to always `NonConfirmable`.

I haven't looked at the v3 branch yet, but I can take a stab at migrating it.